src/main/java/com/autonomic/tmc/auth/exception/SdkServiceException.java

```diff
@@ -0,0 +1,40 @@
+/*-
+ * ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+ * TMC Auth SDK
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Copyright (C) 2016 - 2020 Autonomic, LLC
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ * ______________________________________________________________________________
+ */
+package com.autonomic.tmc.auth.exception;
+
+import static com.autonomic.tmc.auth.exception.ErrorSourceType.SERVICE;
+
+public class SdkServiceException extends BaseSdkException {
+
+    private Integer statusCode = 500;
+
+    public SdkServiceException(String message, Throwable cause) {
+        super(SERVICE, message, cause);
+    }
+
+    public SdkServiceException(String message, int statusCode) {
+        super(SERVICE, message);
+        this.statusCode = statusCode;
+    }
+
```

Comment:
Add an @Override for the get message here, and add the status code, and error Message From the TokenErrorResponse to the end of the message here.

Reply:
Agreed, This is a good way.

Reply:
The override causes the error message to change. Which in turn requires to rewrite existing tests. Added `getStatusCode()` to see errorResponseCode.